### PR TITLE
Fixes deposit flow bugs

### LINKED ISSUE #836: [TASK] Confirm Modal should be closed after a successful Deposit/Withdraw/Transfer
**Is your UI update related to a problem? Please describe.**
When a user does a deposit, withdrawal, or transfer. We'll display a confirm modal for confirming the action. The modal should be closed after a successful Deposit/Withdraw/Transfer.

**Describe the solution you'd like (include image or link to prototype)**
Check if the transaction is successful then close the confirm modal.

CleanShot 2022-12-21 at 22 09 20@2x

### LINKED ISSUE #859: [BUG] Bridge dApp crashes with error page on rejecting a deposit transaction on metamask
### Overview
Rejecting a deposit transaction request on metamask throws `Deposit Transaction Failed` error page. Need a better way to catch and handle this error by not having to show the error page as we are already showing the transaction status in `Transaction Processing` component.

### Error
Screenshot 2023-01-10 at 4 15 43 AM

### Loom video
https://www.loom.com/share/706ef30b990440259185dfdbb4ac1e97

Tested on branch: https://github.com/webb-tools/webb-dapp/pull/850

### LINKED ISSUE #871: [BUG] Wrap & deposit component reopens after dismissing transaction processing component
### Overview
Wrap & deposit component reopens after dismissing transaction processing component (after successful or failed transaction).

### Steps to reproduce

1. After successful or failed transaction, close the `Wrap and Deposit Completed` component
2. Click `DISMISS` button in `Transaction processing `component
3. `Wrap & deposit component` reopens

### Loom video
https://www.loom.com/share/75216830b8784e54a380ef122a3954a3

Tested on branch https://github.com/webb-tools/webb-dapp/pull/850 merged into develop.

### LINKED ISSUE #872: [BUG] Wrap and deposit closing and reopening issue on transaction completion
### Overview
Wrap & deposit component does not close upon dismissing the transaction processing component(after successful or failed transaction) and after manually closing wrap & deposit component, it reopens after few seconds.

### Steps to reproduce

1. After successful or failed transaction, dismiss the `Transaction processing` component
2. `Wrap and Deposit Completed` does not close
3. Manually close the `Wrap and Deposit Completed` component
4. After few seconds of inactivity `Wrap and Deposit` component reopens

### Loom video
https://www.loom.com/share/b87a38bf99c1471b81de9c119abe8f18

Tested on branch https://github.com/webb-tools/webb-dapp/pull/850 merged into develop.

### LINKED ISSUE #874: [BUG] Unable to select moonbase alpha as a source chain to deposit from
### Overview
Unable to select moonbase alpha as a source chain to make a deposit from.

### Steps to reproduce

1. Select `Deposit Tab`
2. Select `Source Chain` input field
3. Select `Moonbase Alpha`
4. `Chain Selection Component` will be closed and the previously connected chain will be selected in `Source Chain` input field

### Loom video
https://www.loom.com/share/60ec95adfcc644188e75196a1fefd395

Tested on branch https://github.com/webb-tools/webb-dapp/pull/850 merged into develop.

### LINKED ISSUE #875: [BUG] Unable to select token for depositing despite having balance in the wallet
### Overview
Unable to select token for depositing despite having balance in the wallet. Ex: Having a balance of some matic tokens in wallet, was unable to select the token for depositing.

### Steps to reproduce

1. Select Deposit Tab
2. Select Source Chain field
3. Select Mumabi
4. Select Token field
5. Try to select MATIC - Make sure you have some matic tokens in your wallet. But this action cannot be performed since MATIC will be disabled showing it's balance as 0

### Loom video
https://www.loom.com/share/70a5040a34e348a4b697e3334b1cb1d0

Tested on branch https://github.com/webb-tools/webb-dapp/pull/850 merged into develop.

### LINKED ISSUE #882: [TASK] Missing back button in Upload Spend Note component
### Overview
`Upload Spend Note` component is missing `Back` button to go back and re-upload the note. And after a note is uploaded `Click to upload or drag & drop` component should be closed and a `Back` button should be provided.

Expected:
Screenshot 2023-01-12 at 3 49 30 AM

Current:
Screenshot 2023-01-12 at 3 47 46 AM

Tested on develop.

### LINKED ISSUE #903: [BUG] New Transaction button does not take user to deposit / transfer / withdraw view
## Current Behaviour 

In the current bridge dApp, when a user initiates a transaction (e.g. deposit, withdraw) the 'New Transaction' button is present. Upon selecting the button the 'Confirm' page is presented to viewers.

## Expected Behaviour 

When users select the 'New Transaction' button it should present the user with the initial deposit / transfer / withdraw view. Whichever operation they were making. 

### Loom link showing issue

https://www.loom.com/share/eb696b2e9817442fae926fc65ee5f43c

### LINKED ISSUE #904: [TASK] Wrapped component shows balance on two lines
## Overview
In the deposit and wrap, and withdraw and unwrap we display the users balance on 2 lines instead of one. The display looks inconsistent, and should be contained to a single line. 

### Screenshot 

ui

### Task Checklist

- [ ] Fix the UI to be on single line
- [x] Remove token composition (e.g.  WebbETH/ETH) to only display bridged asset (e.g. WebbETH) in the above example we should be removing ETH

### PR DESCRIPTION
**Pull request by devpavan04**
## Overview

- New Transaction button now takes user to deposit / transfer / withdraw view.

## Changes video

https://user-images.githubusercontent.com/53374218/213514034-975c0959-fd28-4387-b924-9c91bf7167f4.mov

## Overview

- Wrap & deposit component now closes upon dismissing the transaction processing component(after successful or failed transaction) and does not reopen after the wrap & deposit component is closed.

- Wrap & deposit component does not reopen after dismissing transaction processing component (after successful or failed transaction).
## Changes video

https://user-images.githubusercontent.com/53374218/213514760-62a72d0a-5d72-4901-809f-9a977eac1715.mov

## Overview

- Bridge dApp now does not crash with error page on rejecting a deposit transaction on metamask.
## Changes video

https://user-images.githubusercontent.com/53374218/213515810-fe5fcd92-434b-4dfc-b9df-d21a9928abfb.mov

## Overview

- Moonbase alpha can now be selected the source chain. If user's metamask does not have moonbase network config, metamask is prompted to add the network config.
## Changes video

https://user-images.githubusercontent.com/53374218/213516424-d4b6485d-29eb-4335-ad29-94d71df03631.mov

## Overview

- Users can now select matic token to deposit.
## Changes video

https://user-images.githubusercontent.com/53374218/213517293-8cca92c4-5567-4519-8a05-b7e873bbebfc.mov

## Overview

- Removes token composition (e.g. WebbETH/ETH) to only display bridged asset (e.g. WebbETH)
## Changes screenshot
Screenshot 2023-01-19 at 12 23 30 PM

## Overview

- Upload spend note component now has a back button to go back and re-upload the note.
- After a note gets uploaded `Click to upload or drag & drop` component now closes and a back button should is provided to go back and re-upload the note.
## Changes video

https://user-images.githubusercontent.com/53374218/213723340-dc0421f8-9bfa-4d43-8fea-ebe6f7575a58.mov

### Reference issues to close (if applicable)
- Closes #903
- Closes #875 
- Closes #874 
- Closes #871 
- Closes #872 
- Closes #836 
- Closes #859 
- Closes #904 
- Closes #882

### CODE REVIEW

**Comment by dutterbutter** [comment]
@devpavan04 please run `yarn:format` to resolve ci

**Comment by dutterbutter** [comment]
@devpavan04 any chance you can squeeze https://github.com/webb-tools/webb-dapp/issues/904 in this PR?

File: libs/web3-api-provider/src/webb-provider/bridge-api.ts [hunk 1]
```diff
@@ -30,6 +30,8 @@ export class Web3BridgeApi extends BridgeApi<WebbWeb3Provider> {
       this.inner.getEthersProvider().getSigner()
     );
     const allTokenAddresses = await fungibleToken.contract.getTokens();
+    console.log(allTokenAddresses);
```
Comment (dutterbutter) [inline libs/web3-api-provider/src/webb-provider/bridge-api.ts:33]:
```suggestion
```

File: apps/bridge-dapp/src/containers/DepositContainer/DepositContainer.tsx [hunk 2]
```diff
@@ -72,6 +72,10 @@ export const DepositContainer = forwardRef<
       MainComponentVariants | undefined
     >(undefined);
 
+    const setMainComponentHandler = (component: string | undefined) => {
+      setMainComponentName(component as MainComponentVariants);
+    };
```
Comment (AtelyPham) [inline apps/bridge-dapp/src/containers/DepositContainer/DepositContainer.tsx:77]: It's a bit confusing here. Why did you define the `component: string | undefined` and then cast the variable to type `MainComponentVariants`?

I saw that you use this function to reset the `mainComponentName` to `undefined` so consider using something like this.

```jsx

const resetMainComponent = useCallback(() => {
	setMainComponentName(undefined);
}, [setMainComponentName]);

```

File: apps/bridge-dapp/src/containers/DepositContainer/types.d.ts [hunk 3]
```diff
@@ -41,4 +41,5 @@ export interface DepositConfirmContainerProps extends PropsOf<'div'> {
    * */
   wrappingFlow: boolean;
   wrappableTokenSymbol?: string;
+  setMainComponentHandler: (x: string | undefined) => void;
```
Comment (AtelyPham) [inline apps/bridge-dapp/src/containers/DepositContainer/types.d.ts:44]: Could you consider updating this function as I have suggested?

File: libs/dapp-config/src/currencies/currency-config.ts [hunk 4]
```diff
@@ -250,7 +255,7 @@ export const currenciesConfig: Record<number, CurrencyConfig> = {
   },
   [CurrencyId.moonDEV]: {
     name: 'moonbase Dev',
-    symbol: 'moonDEV',
+    symbol: 'DEV',
```
Comment (AtelyPham) [inline libs/dapp-config/src/currencies/currency-config.ts:258]: We already have the `DEV` token (in local evm), why do you want to update this?

**Review by AtelyPham — COMMENTED** [review]
A couple of suggestions @devpavan04.

File: libs/dapp-config/src/currencies/currency-config.ts (hunk 4 above)
Comment (devpavan04) [reply libs/dapp-config/src/currencies/currency-config.ts:258]: When `wallet_addEthereumChain()` is called to add `Moonbase Alpha`, `Metamask` throws error which says `nativeCurrency.symbol` can have `2-6` character string. `moonDEV` is 7. My bad didn't notice we were already using `DEV`.

Comment (dutterbutter) [reply libs/dapp-config/src/currencies/currency-config.ts:258]: @devpavan04 you will still need to update this value to something unique (e.g. `MDEV`)

**Pull request merged**